### PR TITLE
server: Terminate SASL exchange with proper response on "*"

### PR DESCRIPTION
**Pull request by foxcpp**

### CODE REVIEW

File: status.go [hunk 1]
```diff
@@ -118,3 +118,16 @@ func (r *StatusResp) WriteTo(w *Writer) error {
 
 	return w.writeCrlf()
 }
+
+// ErrStatusResp can be returned by a Handler to replace the default status
```
Comment (emersion) [inline status.go:122]: Handler is a server thing. Maybe we can document that `imap.ErrStatusResp` can be used in `server.Handler` instead?

File: status.go [hunk 2]
```diff
@@ -118,3 +118,16 @@ func (r *StatusResp) WriteTo(w *Writer) error {
...
+// response. The response tag must be empty.
+//
+// To suppress default response, Resp should be set to nil.
+type ErrStatusResp struct {
+	// Response to send instead of default.
+	Resp *StatusResp
+}
+
+func (err *ErrStatusResp) Error() string {
+	return ""
```
Comment (emersion) [inline status.go:132]: Unrelated to this PR, but this is a little bit weird. Maybe we should change this.

File: status.go (hunk 1 above)
Comment (foxcpp) [reply status.go]: server.Handler.Handle code already references ErrStatusResp. I updated it to reference imap.ErrStatusResp and also fixed ErrStatusResp doc to reference server.Handler. Is that better?

File: status.go (hunk 2 above)
Comment (foxcpp) [reply status.go:132]: I am not entirely sure about contexts where StatusResp is used.
Would this work fine?
```go
func (err *ErrStatusResp) Error() string {
	if err.Resp == nil {
		return "imap: suppressed response"
	}
	return err.Resp.Info
}
```

Comment (emersion) [reply status.go:132]: Yeah, this looks good to me. It's just to avoid surprising and hard-to-debug error messages.

File: status.go (hunk 1 above)
Comment (emersion) [reply status.go]: Looks good!

**Comment by emersion** [comment]
Can you split the changes in logical commits, removing the fixup commits?

**Comment by foxcpp** [comment]
> Can you split the changes in logical commits, removing the fixup commits?

Done.

**Pull request merged**